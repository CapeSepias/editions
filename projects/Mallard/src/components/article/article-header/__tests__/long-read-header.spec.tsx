import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { LongReadHeader } from '..'

describe('LongReadHeader', () => {
    it('should show a full featured long read header', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <LongReadHeader
                image={{ source: 'media', path: 'path/to/image.png' }}
                headline="Theresa May makes final speech as PM"
                standfirst="Rolling coverage of the day’s political developments as they happen"
                byline="Andrew Sparrow"
                kicker="News"
            />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should show long read header without an image or a kicker', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <LongReadHeader
                headline="Theresa May makes final speech as PM"
                standfirst="Rolling coverage of the day’s political developments as they happen"
                byline="Andrew Sparrow"
            />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
