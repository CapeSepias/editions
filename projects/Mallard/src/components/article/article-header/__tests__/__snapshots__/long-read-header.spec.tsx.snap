// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LongReadHeader should show a full featured long read header 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexShrink": 0,
        "height": 500,
        "justifyContent": "flex-end",
        "marginTop": -10,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://d2cf1ljtg904cv.cloudfront.net/issue/media/phone/media/path/to/image.png",
      }
    }
    style={
      Array [
        Object {
          "width": "100%",
        },
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#121212",
          "marginEnd": 24,
          "padding": 12,
          "paddingVertical": 6,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderBottomWidth": 0.5,
          "marginBottom": 3,
          "paddingBottom": 18,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "flexShrink": 0,
              "fontFamily": "GTGuardianTitlepiece-Bold",
              "fontSize": 18,
              "lineHeight": 18,
            },
            Object {
              "color": "#ffffff",
            },
          ]
        }
      >
        News
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#121212",
            "flexShrink": 0,
            "fontFamily": "GHGuardianHeadline-Regular",
            "fontSize": 28,
            "lineHeight": 32,
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      Theresa May makes final speech as PM
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#121212",
          "marginEnd": 24,
          "padding": 12,
          "paddingVertical": 6,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "flexShrink": 0,
            "fontFamily": "GuardianTextEgyptian-Reg",
            "fontSize": 17,
            "lineHeight": 21,
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      Rolling coverage of the day’s political developments as they happen
    </Text>
    <RNSVGSvgView
      bbHeight={16}
      bbWidth="100%"
      fill="none"
      height={16}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          null,
          Object {
            "flex": 0,
            "height": 16,
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup
        fill={null}
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      >
        <RNSVGRect
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          rx={0}
          ry={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="100%"
          x={0}
          y={0}
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          rx={0}
          ry={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="100%"
          x={0}
          y={4}
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          rx={0}
          ry={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="100%"
          x={0}
          y={8}
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          rx={0}
          ry={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="100%"
          x={0}
          y={12}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        Array [
          Object {
            "flexShrink": 0,
            "fontFamily": "GuardianTextEgyptian-Bold",
            "fontSize": 17,
            "lineHeight": 21,
          },
          Array [
            Object {
              "color": "#121212",
            },
            Object {
              "color": "#ffffff",
            },
          ],
        ]
      }
    >
      Andrew Sparrow
    </Text>
  </View>
</View>
`;

exports[`LongReadHeader should show long read header without an image or a kicker 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexShrink": 0,
        "height": 500,
        "justifyContent": "flex-end",
        "marginTop": -10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#121212",
          "marginEnd": 24,
          "padding": 12,
          "paddingVertical": 6,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#121212",
            "flexShrink": 0,
            "fontFamily": "GHGuardianHeadline-Regular",
            "fontSize": 28,
            "lineHeight": 32,
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      Theresa May makes final speech as PM
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#121212",
          "marginEnd": 24,
          "padding": 12,
          "paddingVertical": 6,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "flexShrink": 0,
            "fontFamily": "GuardianTextEgyptian-Reg",
            "fontSize": 17,
            "lineHeight": 21,
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      Rolling coverage of the day’s political developments as they happen
    </Text>
    <RNSVGSvgView
      bbHeight={16}
      bbWidth="100%"
      fill="none"
      height={16}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          null,
          Object {
            "flex": 0,
            "height": 16,
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup
        fill={null}
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      >
        <RNSVGRect
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          rx={0}
          ry={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="100%"
          x={0}
          y={0}
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          rx={0}
          ry={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="100%"
          x={0}
          y={4}
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          rx={0}
          ry={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="100%"
          x={0}
          y={8}
        />
        <RNSVGRect
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          height={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          rx={0}
          ry={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          width="100%"
          x={0}
          y={12}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        Array [
          Object {
            "flexShrink": 0,
            "fontFamily": "GuardianTextEgyptian-Bold",
            "fontSize": 17,
            "lineHeight": 21,
          },
          Array [
            Object {
              "color": "#121212",
            },
            Object {
              "color": "#ffffff",
            },
          ],
        ]
      }
    >
      Andrew Sparrow
    </Text>
  </View>
</View>
`;
